main/zoe/guide/contract-installation-and-instance.md

```diff
@@ -0,0 +1,98 @@
+# Contract installation and instance
+
+Zoe has two methods to "create a contract": 
+- `zoe.install(code, moduleFormat)` which returns an `installationHandle`
+- `zoe.makeInstance(installationHandle, issuerKeywordRecord, terms)` which returns an `invite` to participate in the contract
+
+This two-step process to create may feel unnecessarily convoluted at first.
+
+The reason Zoe is designed this way is that it is assumed that the JavaScript source `code` of the contract may be large (several kilobytes, maybe several megabytes) and for the purpose of transparency, this source code will be stored in blockchains when Zoe runs on a blockchain.
+
+```js
+const mySuperContractSourceCode = ` ... 1 Megabyte string ... `;
+const moduleFormat = 'getExport';
+
+const superInstallHandle1 = zoe.install(mySuperContractSourceCode, moduleFormat); // stores 1Mb in the blockchain
+const invite1 = zoe.makeInstance(superInstallHandle1, issuerKeywordRecord, terms); 
+
+const superInstallHandle2 = zoe.install(mySuperContractSourceCode, moduleFormat); // stores another 1Mb in the blockchain 
+// now 2Mb for the same contract
+const invite2 = zoe.makeInstance(superInstallHandle2, issuerKeywordRecord, terms); 
+
+const superInstallHandle3 = zoe.install(mySuperContractSourceCode, moduleFormat); // stores another 1Mb in the blockchain 
```

Comment:
I think having this much code is maybe over-laboring the point, especially since I think the ability to easily check you are using the contract you expect is the more important reason to have `installationHandle`s. Still, this is very persuasive.